plugins/tech-insights-backend/src/service/fact/FactRetrieverRegistry.ts

```diff
@@ -26,16 +26,21 @@ import { ConflictError, NotFoundError } from '@backstage/errors';
  *
  */
 export interface FactRetrieverRegistry {
-  readonly retrievers: Map<string, FactRetrieverRegistration>;
-  register(registration: FactRetrieverRegistration): void;
-  get(retrieverReference: string): FactRetrieverRegistration;
-  listRetrievers(): FactRetriever[];
-  listRegistrations(): FactRetrieverRegistration[];
-  getSchemas(): FactSchema[];
+  register(registration: FactRetrieverRegistration): Promise<void>;
+  get(retrieverReference: string): Promise<FactRetrieverRegistration>;
+  listRetrievers(): Promise<FactRetriever[]>;
+  listRegistrations(): Promise<FactRetrieverRegistration[]>;
+  getSchemas(): Promise<FactSchema[]>;
 }
 
+/**
+ * A basic in memory fact retriever registry.
+ *
+ * You can replace this with a persistance based version using the FactRetrieverRegistry interface.
+ *
+ */
 export class DefaultFactRetrieverRegistry implements FactRetrieverRegistry {
-  readonly retrievers = new Map<string, FactRetrieverRegistration>();
+  private readonly retrievers = new Map<string, FactRetrieverRegistration>();
 
   constructor(retrievers: FactRetrieverRegistration[]) {
     retrievers.forEach(it => {
```

Comment:
There's some async-ness that's being lost here, probably better to handle this in an async static factory instead. Typically we'd do that in a `DefaultFactRetrieverRegistry.create`